lib/s3.py

```diff
@@ -3,26 +3,21 @@
 from botocore.client import Config
 from botocore.exceptions import ClientError
 from lib.logger import logger
-def upload_file_to_s3(bucket: str, filename: str):
+
+
+def upload_file_to_s3_using_filename(bucket: str, filename: str):
     """
     Generic upload helper for s3. Could be moved over to helpers folder...
     """
-    s3_url = os.getenv('S3_ENDPOINT')
-    region = os.getenv('AWS_DEFAULT_REGION')
-    secure = s3_url and s3_url.startswith('https')
-    # Set up the S3 client
-    s3_client = boto3.client('s3',
-        endpoint_url=s3_url,
-        region_name=region,
-        use_ssl=secure)
+    s3_client = get_s3_client()
     # Extract the file name from the local file path
     try:
         s3_client.head_bucket(Bucket=bucket)
     except ClientError as e:
         if e.response['Error']['Code'] == 'NoSuchBucket' or int(e.response['Error']['Code']) in [403, 404]:
             # Create the bucket since it does not exist
             s3_client.create_bucket(Bucket=bucket)
-            logger.info(f'Created bucket {bucket} in MinIO.')
+            logger.info(f'Created bucket {bucket} in MinIO.') # is this accurate? i.e. is this code only for minio?
```

Comment:
I believe the default pattern we established was to create the bucket locally but not in deployed environments, and that the create_bucket would fail because of permissions errors. Tagging @sonoransun to confirm/deny

Reply:
I just updated the log message to be a bit clearer (feel free to push back) -- it now says creating a bucket in s3 (or minio, if running locally)